src/app/components/routemap/RouteVisual.tsx

```diff
@@ -0,0 +1,196 @@
+      return false
+    default:
+      return true
+  }
+}
+
+const AddLines = (props: {
+  rootStatus: string
+  index: number
+  tab: string
+}) => {
+  if (props.index === -1) return
+  const index = lineIndex[props.rootStatus][props.index]
+  return (
+    <RouteLine
+      $ishalfwidth={
+        props.rootStatus !== 'direct' && (index === 2 || index === 3)
+      }
+      className={
+        isPrevStop(props.rootStatus, index, props.tab)
+          ? index === 4
+            ? 'bg-chip-orange'
+            : routeColorTable[props.rootStatus]
+          : 'bg-zinc-200 dark:bg-slate-500'
+      }
+    />
+  )
+}
+
+const AddElements = (props: {
+  rootStatus: string
+  tab: string
+  animationFlagTable: { [key: string]: Array<boolean> }
+}) => {
+  const { t, i18n } = useTranslation()
+  const arrDots: Array<React.JSX.Element> = []
+  const arrChild: Array<React.JSX.Element> = []
+
+  dotIndex[props.rootStatus].forEach((item) => {
+    arrDots.push(
+      <RouteStations key={item}>
+        <Dot
+          className={
+            (isPrevStop(props.rootStatus, item, props.tab)
+              ? item >= 5
+                ? 'bg-chip-orange'
+                : routeColorTable[props.rootStatus]
+              : 'bg-zinc-200 dark:bg-slate-500') +
+            (item === 2 && props.rootStatus === 'yesulin' ? ' opacity-0' : '')
+          }
+        >
+          {item === 3 ? (
+            <SpecialStopsText
+              key={0}
+              lang={i18n.language}
+              className={
+                isPrevStop(props.rootStatus, item, props.tab)
+                  ? routeColorTable[props.rootStatus + 'Text']
+                  : 'text-zinc-200 dark:text-slate-500'
+              }
+            >
+              {props.rootStatus === 'jungang' ? t('jung') : t('yesul')}
+            </SpecialStopsText>
+          ) : (
+            <></>
+          )}
+        </Dot>
+        <AddLines
+          rootStatus={props.rootStatus}
+          index={lineIndex[props.rootStatus].indexOf(item)}
+          tab={props.tab}
+        />
+        <DotAnimation
+          isOn={props.animationFlagTable[props.rootStatus][item]}
+          index={item}
+          color={
+            item >= 5 ? 'bg-chip-orange' : routeColorTable[props.rootStatus]
+          }
+          rootStatus={props.rootStatus}
+        />
+      </RouteStations>,
+    )
+  })
+
+  return (
+    <>
+      {arrDots.map((element: React.JSX.Element, index) => {
+        if (index >= 2 && index <= 4 && props.rootStatus !== 'direct') {
+          arrChild.push(element)
+          if (index === 4) {
+            return (
+              <div
+                key={index}
+                className="col-span-2 grid grid-cols-3 w-[75%] place-items-center"
+              >
+                {arrChild.map((element) => {
+                  return element
+                })}
+              </div>
+            )
+          }
+        } else return element
+      })}
+    </>
+  )
```

Comment:
Please do not use Array of Elements to render a component.
Directly render Elements with data-based approach.

ex) See `Shuttle.tsx` or `Subway.tsx` return statements (return filtered.map...)